subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/ArtifactBackedResolvedVariant.java

```diff
@@ -98,4 +126,37 @@ public void collectBuildDependencies(Collection<? super TaskDependency> dest) {
         }
     }
 
+    private static class DownloadArtifactFile implements RunnableBuildOperation, DescribableBuildOperation<ComponentArtifactIdentifier> {
+        private final ResolvedArtifact artifact;
+        private final Map<ResolvedArtifact, Throwable> artifactFailures;
+
+        DownloadArtifactFile(ResolvedArtifact artifact, Map<ResolvedArtifact, Throwable> artifactFailures) {
+            this.artifact = artifact;
+            this.artifactFailures = artifactFailures;
+        }
+
+        @Override
+        public void run() {
+            try {
+                artifact.getFile();
+            } catch (Throwable t) {
+                artifactFailures.put(artifact, t);
+            }
+        }
+
+        @Override
+        public String getDescription() {
+            return "Resolve artifact " + artifact;
+        }
+
+        @Override
+        public ComponentArtifactIdentifier getOperationDescriptor() {
+            return artifact.getId();
```

Comment:
The build scan plugin needs to be able to tell purely from the descriptor what the operation was doing. A ComponentArtifactIdentifier doesn't really say "what" was done, it only says "something was done involving this artifact".

@breskeby is the build scan plugin already depending on the fact that the descriptor is a `ComponentArtifactIdentifier`?